ignore/src/walk.rs

```diff
@@ -1363,6 +1355,18 @@ impl Worker {
             if !descend {
                 continue;
             }
+
+            let readdir = match readdir {
```

Comment:
Could you maybe add a short comment about the ordering here? e.g., Why `readdir` is being handled here instead of above when `work.read_dir()` is called?

Reply:
I think that readdir shuold be handled after the next if (line 1376). just before the iteration on the folder content.

If the the maximum depth has been reached you should I risk to `self.quit_now()`?